iceoryx2-bb/cxx/include/iox2/container/static_string.hpp

```diff
@@ -26,6 +28,19 @@
 namespace iox2 {
 namespace container {
 
+template <uint64_t>
+class StaticString;
+
+template <typename>
+struct IsStaticString : std::false_type { };
+
+template <uint64_t N>
+struct IsStaticString<StaticString<N>> : std::true_type { };
+
+template <typename T, typename ReturnType>
+using IsStaticStringOrCharArray =
+    typename std::enable_if_t<IsStaticString<T>::value || legacy::is_char_array<T>::value, ReturnType>;
```

Comment:
Unlike `IsStaticString` above, this does not actually answer the question whether `T` is a static string or char array, so I don't like the name too much. How about `RequireStaticStringOrCharArray`?